docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,698 @@
+ +----^-----------------------<+-------+
+     ++                        |        
+ +---+-------+            +----+------+ 
+ | Internet  |            | Internet  | 
+ |  Gateway  |            |  Gateway  | 
+ |    Pod    |            |    Pod    | 
+ |  (VRRP)   |            |  (VRRP)   | 
+ +------^----+            +----^------+ 
+       ++                      ++       
++------+------------------------+-----+ 
+|             Layer2 NAD              |
+| (VRRP Multicast to provide HA VIP)  |
++------------------^------------------+
+                   |
+               Interconnect
+ +-----------------v-------------------+
+ |         Layer3 Primary NAD          |
+ +----^-------------^----------^-------+
+     ++            ++         ++
+   +-+--+        +-+--+    +--+-+
+   |Pod |        |Pod |    |Pod |
+   +----+        +----+    +----+
+```
+
+(TBD: Why is the tenant network just not layer2 then?)
+
+### Story 7: Migration of connected tenant workloads from other platforms like NSX into OVN-Kubernetes
+
+**As a cluster admin**, **I manage** isolated tenants on
+my native NSX platform where they are already seggregated into segments
+using networks. Some of these segments are connected through vRouter
+for mutual access. I want to migrate such workloads into OVN-Kubernetes
+platform by leveraging user defined networks. I would need connecting UDNs
+feature to replicate that same workload architecture. The alternative
+of placing those workloads into same UDN and using network policies
+will not work because it is significant architectural change we would
+need to mandate on each tenant since tenant own their networks.
+
+Example: Tenant-frontend has access to Tenant-backend's databaseAPI
+workloads on NSX through vRouter, I want to migrate both these tenants
+into OVN-Kubernetes platform keeping that architecture intact.
+
+### Story 8: Tenant self-connection without admin intervention
+
+**As a tenant owner**, **I want** to connect my workloads which are part
+of two different namespaces (networks) in the same cluster with no
+admin intervention.
+
+Example: If the same tenant owns microservices across 5 namespaces that
+are 5 different networks, there should be provision for this tenant to
+connect their networks together without the requirement for an admin to
+intervene. The reason could be the same as the above admin user stories
+but this story captures the tenant intent with no admin intervention
+
+## Proposed Solution
+
+This section tries to answer: How can an admin declare two (C)UDNs to be connected?
+
+### Config for enabling the feature
+
+The whole feature will be behind a config flag called `--enable-connecting-networks`
+which has to be set to true to get this feature.
+
+### API Naming [WIP]
+
+We think its better to call the API as one of the following:
+
+* VirtualPrivateConnect (VPC)
+* VirtualPrivateNetworkConnect (VPNC)
+
+for easier adoption across end-users suiting to industry standards.
```

Comment:
can you please add some links to the existing features with similar names?

Reply:
added in latest push